### PR TITLE
Allow snapshot_object have condition and writer option

### PR DESCRIPTION
**Pull request by kuenishi**
This commit also aligns behaviour of `snapshot_object` to `snapshot` for future potential extensions of both functions.

### CODE REVIEW

File: chainer/training/extensions/_snapshot.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from chainer.utils import argument
 
 
-def snapshot_object(target, filename, savefun=npz.save_npz, **kwargs):
+def snapshot_object(target, filename, **kwargs):
```
Comment (crcrpar) [inline chainer/training/extensions/_snapshot.py]: To make the documentation more readable, could you change the docstring as follows?

```python
def snapshot_object(target, filename, **kwargs):
    """snapshot_object(target, filename, *, savefun=npz.save_npz, \
condition=None, writer=None, snapshot_on_error=False)

    Returns a trainer extension to take snapshots of a given object.

    ...
```

**Comment by niboshi** [comment]
Made priority high.

File: chainer/training/extensions/_snapshot.py (hunk 1 above)
Comment (crcrpar) [reply chainer/training/extensions/_snapshot.py]: Done

File: chainer/training/extensions/_snapshot.py [hunk 2]
```diff
@@ -4,8 +4,11 @@
 from chainer.utils import argument
 
 
-def snapshot_object(target, filename, savefun=npz.save_npz, **kwargs):
-    """Returns a trainer extension to take snapshots of a given object.
+def snapshot_object(target, filename, **kwargs):
```
Comment (toslunar) [inline chainer/training/extensions/_snapshot.py]: Are you sure to make a no-compat change by forcing an argument to be a keyword?  The function is publicly available as `chainer.training.extensions.snapshot_object`, while this file `chainer/training/extensions/_snapshot.py` looks private.

Comment (kuenishi) [reply chainer/training/extensions/_snapshot.py]: Thanks for good catch. I knew it was public function and caring default behaviour but less careful about custom savefun case. I added another commit, which might seem strange that the default value of `savefun` being `None`, but it preserves previous behaviour when either `None` or some function is passed.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**